### PR TITLE
respect entry-specific library.type when exposing system context

### LINKED ISSUE #13637: Providing __system_context__ when an entry point sets library.type = 'system'
# Bug report

**What is the current behavior?**

When setting `library.type` per entry point, `__system_context__` is not set for SystemJS entry points and compilation sometimes fails.

**If the current behavior is a bug, please provide the steps to reproduce.**

The easiest way to reproduce this is by changing the test in `configCases/target/system-context`. 

The original config:

```js
/** @type {import("../../../../").Configuration} */
module.exports = {
    output: {
	library: {
	    type: "system"
	}
    },
    node: {
	__dirname: false,
	__filename: false
    }
};
```

I would expect the following config to behave the same:

```js
/** @type {import("../../../../").Configuration} */
module.exports = {
    entry: {
        main: {
	    import: "./index.js",
	    library: {
		type: "system"
	    }
	}
    },
    node: {
	__dirname: false,
	__filename: false
    }
};
```

But it causes an error:

```
TypeError: Cannot read property 'type' of undefined

      219 | 				.for(RuntimeGlobals.systemContext)
      220 | 				.tap("RuntimePlugin", chunk => {
    > 221 | 					if (compilation.outputOptions.library.type === "system") {
          | 					                                      ^
      222 | 						compilation.addRuntimeModule(
      223 | 							chunk,
      224 | 							new SystemContextRuntimeModule()

      at lib/RuntimePlugin.js:221:44
      at Hook.eval (eval at create (node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:13:16)
      at Hook.CALL_DELEGATE [as _call] (node_modules/tapable/lib/Hook.js:14:14)
      at Compilation.processRuntimeRequirements (lib/Compilation.js:2848:7)
      at lib/Compilation.js:2477:12
      at lib/Compilation.js:2680:5
      at node_modules/neo-async/async.js:2818:7
      at done (node_modules/neo-async/async.js:3522:9)
      at lib/Compilation.js:2660:7
      at lib/Compilation.js:2748:32
```

If there is another `library.type` set under `output` then the compilation is successful, but `__system_context__` is undefined when running the application.

**What is the expected behavior?**

Compilation does not fail when `output.library.type` is not set. `__system_context__` is available for entry points with `library.type = 'system'`. 

**Other relevant information:**
webpack version:  5.40.0
Node.js version: 14.17.0
Operating System: macOS 11.4
Additional tools:

### PR DESCRIPTION
**Pull request by cjpearson**
**What kind of change does this PR introduce?**

Fixes #13637. The `__system_context__` injection was not working when using entries with different `library.type` values.

**Did you add tests for your changes?**

Yes

**Does this PR introduce a breaking change?**

No

**What needs to be documented once your changes are merged?**

### CODE REVIEW

File: lib/RuntimePlugin.js [hunk 1]
```diff
@@ -218,7 +218,17 @@ class RuntimePlugin {
...
-					if (compilation.outputOptions.library.type === "system") {
+					const { outputOptions } = compilation;
+					const { library: globalLibrary } = outputOptions;
+					const entryOptions = chunk.getEntryOptions();
+					const libraryType =
+						entryOptions &&
+						entryOptions.library &&
+						entryOptions.library.type !== undefined
+							? entryOptions.library.type
+							: globalLibrary.type;
```
Comment (sokra) [inline lib/RuntimePlugin.js:229]:
```suggestion
					const libraryType =
						entryOptions &&
						entryOptions.library !== undefined
							? entryOptions.library.type
							: globalLibrary.type;
```

When a `library` is specified for an entry it overrides the global one, and doesn't merge with it.

**Review by sokra — CHANGES_REQUESTED** [review]

**Comment by sokra** [comment]
Please make sure to sign the CLA: https://cla.js.foundation/webpack/webpack?pullRequest=13654

**Comment by cjpearson** [comment]
Thanks @sokra. I applied your suggestion and rebased. The CLA should be signed as well.

**Comment by cjpearson** [comment]
@alexander-akait Is it possible to re-run the CI? I don't think the failures were related to this change.

**Comment by alexander-akait** [comment]
@cjpearson Please accept CLA

**Review by sokra — APPROVED** [review]

**Comment by sokra** [comment]
PR looks good, can be merge once CLA is signed.

**Comment by cjpearson** [comment]
Thanks! Just signed it.

**Pull request merged**